eventually/tests/unittests/authentication/testforgetpassword.py

```diff
@@ -0,0 +1,71 @@
+"""
+Forget Password test module
+===========================
+"""
+
+import json
+from django.test import TestCase, Client
+from authentication.views import CustomUser
+from utils.jwttoken import create_token
+
+VALID_EMAIL_TOKEN = create_token({"user_id":100, 'email': 'john@email.com'}, 60 * 60)
+INVALID_EMAIL_TOKEN = create_token({"user_id":999, 'email': 'john@email.com'}, 60 * 60)
+INVALID_TOKEN = "eyJ1c2VyX2lkIjoxMDAsImVtYWlsIjoiam9obkBlbWFpbC5"
+
+class TestForgetPassword(TestCase):
+    """ Class for Forget Password tests """
+
+    def setUp(self):
+        """ Set up test browser client """
+        self.browser = Client()
+
+        user = CustomUser.objects.create(id=100, email="john@email.com")
+        user.set_password = "QwErTy12234"
+        user.save()
+
+
+    def test_forget_password_post(self):
+        """ Test for forget password POST function """
+
+        link = '/api/v1/user/forget_password/'
+
+        email_not_exist = json.dumps({"email": "doesnotexist@gmail.com"})
+        email_exist = json.dumps({"email": "john@email.com"})
+
+        expect_400 = self.browser.post(link, {}, content_type='application/json')
+        self.assertEqual(expect_400.status_code, 400)
+
+        expect_400 = self.browser.post(link, email_not_exist, content_type='application/json')
+        self.assertEqual(expect_400.status_code, 400)
+
+        expect_200 = self.browser.post(link, email_exist, content_type='application/json')
+        self.assertEqual(expect_200.status_code, 200)
+
+    def test_forget_password_put(self):
+        """ Test for forget password PUT function """
+
+        new_password_valid = json.dumps({"new_password": "QwErTy1234"})
+        new_password_invalid = json.dumps({"password": "QwErTy1234"})
+        link_without_token = '/api/v1/user/forget_password/'
+        link_user_exist = link_without_token + str(VALID_EMAIL_TOKEN)[2:-1]
```

Comment:
remove `str`

Reply:
Token is generated as <bytes> therefore str + bytes gives an error:
type(EMAIL_TOKEN) : <class 'bytes'>